test/TestProviderRound.spec.js

```diff
@@ -97,18 +108,57 @@ contract('ProviderRound', accounts => {
       });
     });
 
+    // TODO: Doc about providers, pool and active pool
+    it('should add the provider to the pool if he is Unregistered and size < maxSize', async () => {
+      // Check that provider isn't registered yet
+      assert.equal(false, await providerRound.containsProvider(accounts[3]));
+      // Check the size of the pool increases by 1
+      let providerPool = await providerRound.providerPool.call();
+      const previousSize = providerPool[3].toNumber(); // [3] is current size of the pool
+      await approveBondProvider(21, 13, 3, 26, 1, accounts[3]);
+      providerPool = await providerRound.providerPool.call();
+      assert.equal(previousSize + 1, providerPool[3]);
+      // Check that the provider is registered in the pool now
+      assert.equal(true, await providerRound.containsProvider(accounts[3]));
+    });
+
+    it('should fail if provider is Unregistered and size == maxSize', async () => {
+      await approveBondProvider(20 ,10, 2, 25, 1, accounts[4]);
+      await assertFail( approveBondProvider(20 ,10, 2, 25, 1, accounts[5]) );
```

Comment:
We should have a check here that gets the maxSize of the pool from the contract and asserts that it is presently maxed out.